Zadatak1.ps1

```diff
@@ -0,0 +1,59 @@
+﻿param(
+    [Parameter(Mandatory=$true)]
+    [string]$dateFolder # Datum kao parametar u forrmatu 'yyyy-MM-dd'
+)
+
+$rootDir = "izvodi"
+
+# Osigurati da izvodi postoje
+if (-Not (Test-Path $rootDir)) {
+    Write-Error "'$rootDir' direktorijum ne postoji."
+    exit
+}
+
+# Prolazimo kroz svaki petocifreni folder
+Get-ChildItem -Path $rootDir -Directory | Where-Object { $_.Name -match '^\d{5}$' } | ForEach-Object {
+    $folder = $_
+    $dateFolderPath = Join-Path -Path $folder.FullName -ChildPath $dateFolder
+    
+    # Nastavlja ako trazeni datum folder postoji
+    if (Test-Path $dateFolderPath) {
+        # Privremeni folder za ekstraktovane fajlove
+        $tempExtractionPath = Join-Path -Path $dateFolderPath -ChildPath "temp_extraction"
+        New-Item -Path $tempExtractionPath -ItemType Directory -Force | Out-Null
+
+        # Pronalazi sve zip fajlove i ekstraktuje ih
+        Get-ChildItem -Path $dateFolderPath -File | Where-Object { $_.Extension -eq '.zip' } | ForEach-Object {
+            $zipFile = $_.FullName
+            $unzipCommand = "unzip -o `"$zipFile`" -d `"$tempExtractionPath`""
+            Invoke-Expression $unzipCommand
```

Comment:
There is no error checking on what unzip did: For example this dummy example 

![image](https://github.com/akecan/powershell-practise/assets/85767/9d7c61bd-8d5f-4525-a4a2-525b46b65141)

Furthermore, since you used Compres-Archive later, why not [Expand-Archive](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.archive/expand-archive?view=powershell-7.4).

Any external dependency such as unzip would have to be in the README.md detailing on how to install it.

Reply:
Furthermore, using external utility `unzip` without any documentation on prerequisites for this script to run isn't usage friendly. More importantly, pwsh has interal arhiver: [Compress-Archive](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.archive/compress-archive?view=powershell-7.40) which was an obvious thing to use here, unless there is specific reason.